Makefile

```diff
@@ -0,0 +1,132 @@
+BACKEND_SERVICES ?= autoignore autostarter configuration database detection fileobserver mitigation notifier prefixtree
+TAP_SERVICES ?= bgpstreamhisttap bgpstreamkafkatap bgpstreamlivetap exabgptap riperistap
+FRONTEND_SERVICES ?= frontend
+
+SERVICES ?= $(BACKEND_SERVICES) + $(TAP_SERVICES) + $(FRONTEND_SERVICES)
+
+PUSH ?= false
+BUILD_TAG ?= latest
+CONTAINER_REPO ?= docker.io/inspiregroup
+
+.PHONY: $(BACKEND_SERVICES)
+$(BACKEND_SERVICES): # build backend container
+$(BACKEND_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest
+ifneq ($(BUILD_TAG), "latest")
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG)
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		backend-services/$@/
+ifeq ($(PUSH), "true")
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: $(TAP_SERVICES)
+$(TAP_SERVICES): # build tap container
+$(TAP_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest
+ifneq ($(BUILD_TAG), "latest")
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG)
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		monitor-services/$@/
+ifeq ($(PUSH), "true")
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: build-backend
+build-backend: # builds all backend containers
+build-backend: $(BACKEND_SERVICES)
+
+.PHONY: build-taps
+build-taps: # builds all tap containers
+build-taps: $(TAP_SERVICES)
+
+.PHONY: build-frontend
+build-frontend: # builds frontend container
+build-frontend:
+	@docker pull $(CONTAINER_REPO)/artemis-frontend:latest
+	@docker build --build-arg revision=$(git rev-parse --short HEAD) -t artemis-frontend:$(BUILD_TAG) --cache-from $(CONTAINER_REPO)/artemis-frontend:latest --cache-from $(CONTAINER_REPO)/artemis-frontend:$(BUILD_TAG) frontend/
```

Comment:
be careful we are using `$(CONTAINER_REPO)/artemis-tempfrontend` so that we do not mess with the other old frontend (I have made changes for modularization which we should keep until the next release). See the travis.yml we use in master as well as the `docker-compose.yaml` `frontend` section.

Reply:
the local image is still `frontend`, but the remote one in the repo is on purpose called `tempfrontend`

Reply:
I am reopening this since not resolved (so that we do not forget it)